action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -557,6 +557,14 @@ func (stateDB *StateDBAdapter) RevertToSnapshot(snapshot int) {
 	// restore gas refund
 	if !stateDB.manualCorrectGasRefund {
 		stateDB.refund = stateDB.refundSnapshot[snapshot]
+		delete(stateDB.refundSnapshot, snapshot)
+		for i := snapshot + 1; ; i++ {
+			if _, ok := stateDB.refundSnapshot[i]; ok {
+				delete(stateDB.refundSnapshot, i)
+			} else {
+				break
+			}
+		}
```

Comment:
in this fix: https://github.com/iotexproject/iotex-core/commit/fcb8d7b5ea40b04744b82fcf054695c1c12720ba, we need to clear snapshot >= n, for `RevertToSnapshot(n)`